### ISSUE TITLE
Using \setTestFont with weight parameter crashes class

### ISSUE THREAD

**Issue by alerque** [sile-typesetter/fontproof#41]
Extracted from @n7s's [comment](https://github.com/silnrsi/font-andika-mtihani/pull/2#issuecomment-1493598409):

> With the new 2.0.1-1 version installed (as verified in
> /usr/local/bin/fontproof targettting version 5.2 of lua as AFAICT the Ubuntu
> package pulls in a version of sile targetting version 5.2) I get a new error:
>
> ```
> [builder@smith-jammy]  🌱 master origin/master (b836aba) +/-  ? 
> ⬢ [Docker] /smith/font-andika-mtihani/results 🐳 🍏 ❯  fontproof --filename="AndikaMtihani-Regular.ttf" ../tests/Latin.sil 
> SILE v0.14.8 (Lua 5.2)
> <../tests/Latin.sil> as sil
> 
> ! Attempted to get an undeclared class option 'weight' at ../tests/Latin.sil: in \font near 3:1: in \setTestFont[features="+clig,+kern",papersize="a4",size="18pt"]
> 
> error summary:
> 	Processing at: ../tests/Latin.sil: in \font near 3:1: in \setTestFont[features="+clig,+kern",papersize="a4",size="18pt"]
> 	Using code at: /usr/share/sile/classes/base.lua:34: Attempted to get an undeclared class option 'weight'
> 
> Run with --traceback for more detailed trace leading up to errors.
> ```
> 
> It looks like \setTestFont parameters are the issue. How would I debug this
> further? (not getting a backtrace with --traceback). A project like Gentium
> has more .sil files for further testing.

**Comment by alerque** [comment]
I started setting class options from `\setFontOptions` assuming everything passed in there is things that might also be set on the class or from the CLI. Evidently I assumed that since `--filename` would be the primary use case selectors like `weight=` and `style=` were not necessary. The solution will be to add those as class options (and CLI flags).

**Comment by alerque** [comment]
Go ahead and give this a shot from the DEV rockspec, I think this should be fixed on *master*. It it is we'll tag a release.

**Comment by n7s** [comment]
Thanks, I can confirm that version rockspec 2.0.2-1 now works as intended (SILE v0.14.8 - Lua 5.2).

**Comment by n7s** [comment]
See the[ integration change](https://github.com/silnrsi/smith/commit/1db0a9ce7cd1e5f5ddddf97de42894388f69e1a7) in smith, our toolchain. We only pass --filename and --output parameters, we expect all the other options to be set in the .sil document.

**Comment by n7s** [comment]
BTW trying to install and then use the dev rockspec via `sudo luarocks install --server=https://luarocks.org/dev fontproof`  I get the following error: `Using code at: /usr/share/sile/classes/base.lua:143: module 'packages.fontproof' not found`. But the non-dev version works, not sure what subtlety I'm missing. Shouldn't the dev do an install directly from source?

**Comment by alerque** [comment]
I would have expected the usage to be:

``` console
$ luarocks install fontproof dev
```

... and for it to work exactly the same as installing a tagged version. I'll try to get on a system where I can test that later.

**Comment by alerque** [comment]
I was not able to replicate your error. That being said the way I was testing with a local (non system root) install also didn't work like I thought it might:

``` console
$ cd foo
$ luarocks --local --tree . install fontproof --server=https://luarocks.org/dev
$ eval $(luarocks --local --tree . path)
$ fontproof -t full
SILE v0.14.8 (Lua 5.4)

! Unable to find file 'templates/full.sil': no file 'templates/full.sil'
...
```

It seems there is still some gremlin in path handling. It works okay if I do a local install from `make install` in the source tree, but not on an independent system.

I don't want to do a root install of *anything* using LuaRocks on these systems so I'll have to spin up a VM to look into what is ending up where.

**Comment by alerque** [comment]
Okay my error is a different issue, and I see what that is about. I lost some code I wrote for path handling in a refactor somewhere.

I've also replicated the error show by you, and yes the DEV packaging behaves differently for some reason.

**Comment by alerque** [comment]
My (unrelated) issue should be fixed in 39c1527. I pushed it to master directly because it's hard to test the dev rockspec without actually installing it, which it does from Git. Still working on other error.

**Comment by alerque** [comment]
@n7s Can you try again with the dev rockspec? It seems to be working for me now but I'm not sure *why*. I think maybe the last iteration of rockspec changes didn't actually get published like we thought it did, so we were installing new code via a miss-matched rockspec. Now since I just touched it for other reasons it republished and now works. I suspect you'll find it works too.

If so I'll publish another patch release to cover the tweaks we've made.

**Comment by n7s** [comment]
The dev rockspec works now for me as well. Thank you!

$ fontproof --version
SILE v0.14.8 (Lua 5.2)
FontProof installed from rockspec dev-1

Could it have been something to do with the lua version? (I always only tested on 5.2 from a .deb package and not with sile from source which I expect might built against newer lua when available).

**Comment by alerque** [comment]
No, Lua versions wasn't the problem. We had updated the DEV rockspec here in the repository but it did not get published to LuaRocks properly. Hence we were installing the newest version of code in the repository using an old version of the manifest and hence things we were getting new code placed in old locations.

The deal is the rockspecs get pushed automatically when updated here, but **only** if they are touched in the last commit. I pushed two or more commits at once and the rockspec change was not in the most recent one, so it didn't detect it as needing publishing.
